.github/workflows/deploy-to-vercel.yml

```diff
@@ -0,0 +1,58 @@
+name: Vercel Deployment
+
+on:
+  workflow_call:
+    inputs:
+      vercel-environment:
+        required: true
+        type: string
+    secrets:
+      vercel-token:
+        required: true
+      vercel-org-id:
+        required: true
+      vercel-project-id:
+        required: true
+      github-token:
+        required: true
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Vercel CLI
+        run: npm install --global vercel@latest
+        shell: bash
+
+      - name: Pull Vercel Environment Information
+        run: vercel pull --yes --environment=${{ inputs.vercel-environment }} --token=${{ secrets.vercel-token }}
```

Comment:
nit: I wonder if we could re-export the `secrets.vercel-token` once to an env so we don't need to pass `--token=` to every command. I'm not sure if Vercel's CLI takes a token env though, but I feel it should